### PR TITLE
Atomic operations interface cleanup & atomicCompareExchange function fix

### PR DESCRIPTION
**Pull request by ywwu928**
# Description

## PR description

This PR _write a high-level description of your pull request_

## Checklist

- [v] The additions follow the code standards in the [developer guide](pando-rt/docs/developer.md).
- [v] New or existing tests cover these changes.
- [v] The documentation is up to date with these changes.

### CODE REVIEW

File: microbench/bfs/src/sssp.cpp [hunk 1]
```diff
@@ -12,9 +12,7 @@ void bfs::updateData(std::uint64_t val, pando::GlobalRef<std::uint64_t> ref) {
     if (val >= temp) {
       break;
     }
-  } while (!pando::atomicCompareExchange(&ref, pando::GlobalPtr<std::uint64_t>(&temp),
-                                         pando::GlobalPtr<std::uint64_t>(&val),
-                                         std::memory_order_relaxed, std::memory_order_relaxed));
+  } while (!pando::atomicCompareExchange(&ref, temp, val));
```
Comment (AdityaAtulTewari) [inline microbench/bfs/src/sssp.cpp:15]: why is this change required if the below is what is causing the problem?

**Review by AdityaAtulTewari — CHANGES_REQUESTED** [review]

File: microbench/bfs/src/sssp.cpp (hunk 1 above)
Comment (ywwu928) [reply microbench/bfs/src/sssp.cpp:15]: I mean it certainly is not necessary but it saves us a bunch of transformation. Now I don't have to transform a pointer to uint64_t to GlobalPtr and dereference it within the atomicCompareExchange function. I stick with whatever uint64_t value for the entire function.

**Review by AdityaAtulTewari — APPROVED** [review]
